pkg/odo/cli/registry/registry.go

```diff
@@ -0,0 +1,226 @@
+package registry
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"github.com/redhat-developer/odo/pkg/log"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	odoutil "github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/registry"
+	"github.com/redhat-developer/odo/pkg/util"
+	"github.com/spf13/cobra"
+)
+
+const RecommendedCommandName = "registry"
+
+var Example = `  # Get all devfile components
+  %[1]s
+
+# Filter by name
+%[1]s --filter nodejs
+
+# Filter by name and devfile registry
+%[1]s --filter nodejs --devfile-registry DefaultDevfileRegistry
+
+# Show more details
+%[1]s --details`
```

Comment:
Could you add the example with  `--details --devfile <name> --devfile-registry <registry>` to get the details of a named devfile?